source/Detector/Amptek/AmptekSDD123DetectorManager.cpp

```diff
@@ -250,8 +248,13 @@ void AmptekSDD123DetectorManager::onSpectrumEventReceived(AmptekSpectrumEvent *s
 		if (presetDwellLocalStartTime_.isValid())
 			elapsedTime = ((double)presetDwellLocalStartTime_.msecsTo(presetDwellLocalEndTime_))/1000;
 
+		AMDSDwellSpectralDataHolder *dwellHistogram = new AMDSDwellSpectralDataHolder(detector_->dataType(), detector_->bufferSize(), this);
```

Comment:
Are you not doing anything with this AMDSDwellSpectralDataHolder?  It looks like you create it and set it but then don't use it anywhere.